phpBB/phpbb/legacy/http/legacy_handler.php

```diff
@@ -0,0 +1,276 @@
+		$url = $this->request->server->get('REDIRECT_SCRIPT_URL');
+		$this->request->server->set('SCRIPT_NAME', $url);
+		$this->request->server->set('PHP_SELF', $url);
+		$this->request->server->set('SCRIPT_FILENAME', $this->legacyPath);
+		$this->request->overrideGlobals(); // Do the same in phpBB object
+
+		// Change path to the script's path
+		//chdir(dirname($this->legacyPath));
+
+		// Enable output buffering to be able to fetch the legacy's content
+		// and encapsulate later in a response object.
+		ob_start();
+	}
+
+	/**
+	 * Handles the response
+	 *
+	 * Restores the default error level, and encapsulates the legacy response's
+	 * content in a Symfony response.
+	 *
+	 * @return Response
+	 */
+	public function handleResponse()
+	{
+		// Restore error reporting in Symfony part of the request.
+		$this->switchErrorLevel($this->previousErrorLevel);
+
+		// Fetch content from the legacy (via output buffering).
+		$content = ob_get_clean();
+
+		// Restore the path
+		//chdir($this->basePath);
+		$rawHeaders = array_map(function ($header) {
+			return explode(': ', $header);
+		}, headers_list());
+
+		$headers = [];
+		foreach ($rawHeaders as $header) {
+			list($k, $v) = $header;
+			header_remove($k);
+			$headers[$k] = $v;
+		}
+
+		// Encapsulate legacy response in Symfony
+		$response = new Response($content, http_response_code(), $headers);
+
+		$this->stopwatchStop('legacy');
+
+		// Add additional stuff from Symfony (additional headers, WDT when in dev, etc.)
+		return $this->httpKernel->filterResponse($response, $this->request);
+	}
+
+	/**
+	 * Handles an exception, and wraps it in a Symfony response
+	 *
+	 * @param \Exception $e
+	 * @param Request $request
+	 * @return Response
+	 */
+	public function handleException(\Exception $e, Request $request)
+	{
+		$this->switchErrorLevel($this->previousErrorLevel);
+
+		return $this->httpKernel->handleException($e, $request);
+	}
+
+	/**
+	 * Switches the error level
+	 *
+	 * @param int $errorLevel
+	 * @return int
+	 */
+	private function switchErrorLevel($errorLevel)
+	{
+		$this->previousErrorLevel = error_reporting($errorLevel);
+	}
+
+	/**
+	 * Starts a stopwatch section, when debug mode is activated
+	 *
+	 * @param $name
+	 */
+	private function stopwatchStart($name)
+	{
+		if ($this->debug) {
+			//$this->stopwatch->start($name);
+		}
+	}
+
+	/**
+	 * Stops a stopwatch section, when debug mode is activated
+	 *
+	 * @param $name
+	 */
+	private function stopwatchStop($name)
+	{
+		if ($this->debug) {
+			//$this->stopwatch->stop($name);
+		}
+	}
+
+	/**
+	 * A simple helper to throw a NotFoundHttpException
+	 *
+	 * @param Request $request
+	 * @return Response
+	 */
+	private function handleNotFound(Request $request)
+	{
+		return $this->handleException(new NotFoundHttpException(), $request);
+	}
+
+	public function handleTrailingSlash(Request $request)
+	{
+		$parts = parse_url($request->getUri());
+
+		if(isset($parts['path']) && '/' === substr($parts['path'], -1)) {
```

Comment:
`(! isset($parts['path']) || '/' === substr($parts['path'], -1)) && ! isset($parts['query'])` should be safer